### PR TITLE
feat: add mysql migration support

### PR DESCRIPTION
**Pull request by tengattack**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [ ] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [ ] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [ ] <del>**Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?</del>
- [ ] **Testing:** Have you written and run sufficient tests for validating the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **feat**: Introduces a new feature or enhancement to the codebase

# Changelog Entry

### Description

- [Concisely describe the changes made in this pull request, including any relevant motivation and impact (e.g., fixing a bug, adding a feature, or improving performance)]

Previously, we has no way to use mysql but manually migrate the data store.
So, I have made a change to it, now we can start with mysql from scratch.

https://github.com/open-webui/open-webui/discussions/3903 https://github.com/open-webui/open-webui/discussions/2529

### Added

- Add support for mysql migration

### CODE REVIEW

File: backend/apps/webui/internal/migrations/017_add_user_oauth_sub.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def migrate(migrator: Migrator, database: pw.Database, *, fake=False):
 
     migrator.add_fields(
         "user",
-        oauth_sub=pw.TextField(null=True, unique=True),
+        oauth_sub=pw.CharField(max_length=255, null=True, unique=True),
```
Comment (tjbck) [inline backend/apps/webui/internal/migrations/017_add_user_oauth_sub.py:38]: Not sure how this change is relevant.

**Comment by tjbck** [comment]
I'd like to merge this to make our code more compatible with mysql, but it still won't be added to the list of officially supported database. The rest LGTM!

File: backend/apps/webui/internal/migrations/017_add_user_oauth_sub.py (hunk 1 above)
Comment (tengattack) [reply backend/apps/webui/internal/migrations/017_add_user_oauth_sub.py:38]: MySQL requires the unique field has a key length.

I didn't know the typical `oauth_sub` text size, is there 255 enough? or we should specified the unique key prefix length?

https://stackoverflow.com/questions/1827063/mysql-error-key-specification-without-a-key-length

**Comment by tjbck** [comment]
I might change my mind later in the future, but I'd rather not support MySQL at the moment. Thanks for the PR!

**Pull request closed without merging**